examples/AdvancedOperations/AddSmartDisplayAd.php

```diff
@@ -0,0 +1,434 @@
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignResourceName
+    ) {
+        // Constructs an ad group and set its type.
+        $adGroup = new AdGroup([
+            'name' => new StringValue(['value' => 'Earth to Mars Cruises #' . uniqid()]),
+            'campaign' => $campaignResourceName,
+            'status' => AdGroupStatus::PAUSED,
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Issues a mutate request to add the ad group.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        /** @var MutateAdGroupsResponse $adGroupResponse */
+        $adGroupResponse = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        // Print out some information about the added ad group.
+        $adGroupResourceName = $adGroupResponse->getResults()[0]->getResourceName();
+        printf("Added ad group named '%s'.%s", $adGroupResourceName, PHP_EOL);
+
+        return $adGroupResourceName;
+    }
+
+    /**
+     * Creates a responsive display ad, which is a recommended ad type for Smart Display campaigns.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @param string $adGroupResourceName the ad group resource name
+     * @param string|null $marketingImageAssetResourceName optional, the resource name of marketing
+     *     image asset
+     * @param string|null $squareMarketingImageAssetResourceName optional, the resource name of
+     *     square marketing image asset
+     */
+    private static function createResponsiveDisplayAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adGroupResourceName,
+        string $marketingImageAssetResourceName = null,
+        string $squareMarketingImageAssetResourceName = null
+    ) {
+        // Creates a new image asset for marketing image and square marketing image if there is no
+        // assets' resource names specified.
+        $marketingImageAssetResourceName = $marketingImageAssetResourceName ?:
+            self::createImageAsset(
+                $googleAdsClient,
+                $customerId,
+                self::MARKETING_IMAGE_URL,
+                'Marketing Image'
+            );
+        $squareMarketingImageAssetResourceName = $squareMarketingImageAssetResourceName ?:
+            self::createImageAsset(
+                $googleAdsClient,
+                $customerId,
+                self::SQUARE_MARKETING_IMAGE_URL,
+                'Square Marketing Image'
+            );
+
+        // Creates an ad group ad with the ad info related to the responsive display ad.
+        $adGroupAd = new AdGroupAd([
+            'ad_group' => $adGroupResourceName,
+            'status' => AdGroupAdStatus::PAUSED,
+            'ad' => new Ad([
```

Comment:
The ad object here is pretty big, so the AdGroupAd creation doesn't fit in a single screen.

How about extracting it to a separate variable?

Reply:
Done.